src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.cpp

```diff
@@ -197,32 +197,62 @@ void VulkanRenderer::DetermineVendor()
 		break;
 	}
 
-	if (IsRunningInWine())
+	VkDriverId driverId = driverProperties.driverID;
+
+	if(driverId == VK_DRIVER_ID_MESA_RADV || driverId == VK_DRIVER_ID_INTEL_OPEN_SOURCE_MESA)
+	{
 		m_vendor = GfxVendor::Mesa;
+	}
+
 
 	forceLog_printf("Using GPU: %s", properties.properties.deviceName);
 
+	m_driverVersion = properties.properties.driverVersion;
+
 	if (m_featureControl.deviceExtensions.driver_properties)
 	{
 		forceLog_printf("Driver version: %s", driverProperties.driverInfo);
 
 		if(m_vendor == GfxVendor::Nvidia)
 		{
 			// multithreaded pipelines on nvidia (requires 515 or higher)
-			m_featureControl.disableMultithreadedCompilation = (StringHelpers::ToInt(std::string(driverProperties.driverInfo)) < 515);
+			m_featureControl.disableMultithreadedCompilation = (std::get<0>(GetDriverVersion()) < 515);
 		}
 	}
 
 	else
 	{
 		forceLog_printf("Driver version (as stored in device info): %08X", properties.properties.driverVersion);
-		
-		if(m_vendor == GfxVendor::Nvidia)
-		{
-			// if the driver does not support the extension,
-			// it is assumed the driver is under version 515
-			m_featureControl.disableMultithreadedCompilation = true;
-		}
+	}
+}
+
+const std::tuple<uint16, uint16, uint16, uint16> VulkanRenderer::GetDriverVersion()
```

Comment:
Do we really need GetDriverVersion()? The only necessary use-case is for the Nvidia pipeline check which is already a solved problem. We really don't want to encourage more conditional logic to be built on top of the driver version, especially since vendors are free to change their versioning and driverID encoding at any moment.

Reply:
removed it